### PR TITLE
[WiP] [FIXED JENKINS-28655 and JENKINS-28656] - Docker Traceability can create image fingerprints on-demand

### PR DESCRIPTION
**Pull request by oleg-nenashev**
On the top of #3 and #5 

JENKINS-28656
- [x] Add the global configuration of DockerTraceabilityPlugin
- [x] Adjust the logic to create fingerprints on-demand
- [x] Unit test
- [ ] Update docs

JENKINS-28655
- [x] - New item skeleton with the automatic injection
- [x] - Better UI - show item references on run summaries
- [x] - Automatically setup build descriptions 

Open questions:
- [ ] - TBD - retention policies in Docker Reference Job (remove buttons, etc.)

@reviewbybees (esp. @jtnord )

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/docker/traceability/DockerTraceabilityPlugin.java [hunk 1]
```diff
@@ -55,10 +62,49 @@
...
+        }
+        return plugin;
+    }
+    
+    @Override
+    protected void load() throws IOException {
+        super.load();
+        if (configuration == null) {     
+            configuration = DockerTraceabilityPluginConfiguration.getDefault();        
+            save();
```
Comment (jtnord) [inline src/main/java/org/jenkinsci/plugins/docker/traceability/DockerTraceabilityPlugin.java]: why save the default config?

File: src/main/java/org/jenkinsci/plugins/docker/traceability/DockerTraceabilityPlugin.java [hunk 2]
```diff
@@ -55,10 +62,49 @@
...
+    }   
+
+    public @Nonnull DockerTraceabilityPluginConfiguration getConfiguration() {
+        return configuration;
     }
-    
 
+    @Override 
+    public void start() throws Exception {
+        load();
```
Comment (jtnord) [inline src/main/java/org/jenkinsci/plugins/docker/traceability/DockerTraceabilityPlugin.java]: so if someone calls getConfiguration before start() then a `@Nonnull` field will be null?

File: src/main/java/org/jenkinsci/plugins/docker/traceability/DockerTraceabilityPluginConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+import javax.annotation.Nonnull;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+/**
+ * Configuration of {@link DockerTraceabilityPlugin}.
+ * @author Oleg Nenashev
+ */
+public class DockerTraceabilityPluginConfiguration implements Describable<DockerTraceabilityPluginConfiguration> {
+    
+    private static final DockerTraceabilityPluginConfiguration DEFAULT =
```
Comment (jtnord) [inline src/main/java/org/jenkinsci/plugins/docker/traceability/DockerTraceabilityPluginConfiguration.java]: why don't you just initialize the fields with their defaults?

**Comment by jtnord** [comment]
@oleg-nenashev can you merge the code cleanup PR - there seems to be a lot of fluff in the diffs that should not be.

File: src/main/java/org/jenkinsci/plugins/docker/traceability/DockerTraceabilityPlugin.java (hunk 2 above)
Comment (oleg-nenashev) [inline src/main/java/org/jenkinsci/plugins/docker/traceability/DockerTraceabilityPlugin.java]: Yes, makes sense to explicitly prevent it

**Comment by oleg-nenashev** [comment]
@jtnord 
I would be happy to do it as well, but I need thumbs up in #3

File: src/main/java/org/jenkinsci/plugins/docker/traceability/DockerTraceabilityPluginConfiguration.java (hunk 3 above)
Comment (oleg-nenashev) [inline src/main/java/org/jenkinsci/plugins/docker/traceability/DockerTraceabilityPluginConfiguration.java]: Such approach is useful when you decide to implement the <code>resetDefaults()</code> call at some point

**Comment by oleg-nenashev** [comment]
I'll reopen the PR

File: src/main/java/org/jenkinsci/plugins/docker/traceability/DockerTraceabilityPlugin.java (hunk 2 above)
Comment (oleg-nenashev) [inline src/main/java/org/jenkinsci/plugins/docker/traceability/DockerTraceabilityPlugin.java]: done in #8

**Pull request closed without merging**